src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -165,6 +176,82 @@ private StaticLangXApiView generateXApi(SurfaceTransformerContext context) {
     return xapiClass.build();
   }
 
+  private List<StaticLangPagedResponseView> generatePagedResponseWrappers(
+      SurfaceTransformerContext context) {
+    List<StaticLangPagedResponseView> pagedResponseWrappers = new ArrayList<>();
+
+    for (Method method : context.getSupportedMethods()) {
+      MethodTransformerContext methodContext = context.asMethodContext(method);
+      MethodConfig methodConfig = context.getMethodConfig(method);
+      ModelTypeTable typeTable = context.getTypeTable();
+
+      if (methodConfig.isPageStreaming()) {
+        StaticLangPagedResponseView.Builder pagedResponseWrapper =
```

Comment:
This building part is quite long. Maybe worth creating a helper method - generatePagedResponseWrapper(MethodTransformerContext context)